packages/@lwc/compiler/src/transformers/template.ts

```diff
@@ -83,6 +84,7 @@ function serialize(
     buffer += 'if (_implicitStylesheets || _implicitScopedStylesheets) {\n';
     buffer += `  tmpl.stylesheetToken = "${scopeToken}"\n`;
     buffer += '}\n';
+    buffer += 'freezeTemplate(tmpl);\n';
```

Comment:
This does not strictly need to be added to the ACT compiler, since all it does it log a warning. Maybe someday when this actually freezes, we can add this to the ACT compiler.

In either case, it'd be just a one-line code change to the ACT compiler.